qiime/sdk/plugin_manager.py

```diff
@@ -65,6 +66,7 @@ def _integrate_plugin(self, plugin):
                                  % transformer_record)
             self.transformers[input][output] = transformer_record
 
+        self.formats.update(plugin.formats)
```

Comment:
Instead of doing an `update`, can you check for duplicate format registrations here? That seems more likely to happen than a duplicate registration within a plugin, though that check you have is great to keep too.